include/sparrow/fixed_size_layout.hpp

```diff
@@ -0,0 +1,370 @@
+    {
+        assert(i < size());
+        return data()[i];
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::operator[](size_type i) -> reference
+    {
+        assert(i < size());
+        return reference(value(i), has_value(i));
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::operator[](size_type i) const -> const_reference
+    {
+        assert(i < size());
+        return const_reference(value(i), has_value(i));
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::bitmap() -> bitmap_range
+    {
+        return std::ranges::subrange(bitmap_begin(), bitmap_end());
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::values() -> value_range
+    {
+        return std::ranges::subrange(value_begin(), value_end());
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::bitmap() const -> const_bitmap_range
+    {
+        return std::ranges::subrange(bitmap_cbegin(), bitmap_cend());
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::values() const -> const_value_range
+    {
+        return std::ranges::subrange(value_cbegin(), value_cend());
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::has_value(size_type i) -> bitmap_reference
+    {
+        assert(i < size());
+        return m_data.bitmap[i];
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::has_value(size_type i) const -> bitmap_const_reference
+    {
+        assert(i < size());
+        return m_data.bitmap[i];
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::value_begin() -> value_iterator
+    {
+        return value_iterator{data()};
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::value_end() -> value_iterator
+    {
+        return value_iterator{data() + size()};
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::value_cbegin() const -> const_value_iterator
+    {
+        return const_value_iterator{data()};
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::value_cend() const -> const_value_iterator
+    {
+        return const_value_iterator{data() + size()};
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::bitmap_begin() -> bitmap_iterator
+    {
+        return m_data.bitmap.begin();
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::bitmap_end() -> bitmap_iterator
+    {
+        return m_data.bitmap.begin() + size();
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::bitmap_cbegin() const -> const_bitmap_iterator
+    {
+        return m_data.bitmap.cbegin();
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::bitmap_cend() const -> const_bitmap_iterator
+    {
+        return m_data.bitmap.cbegin() + size();
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::data() -> pointer
+    {
+        assert(m_data.buffers.size() > 0);
+        return m_data.buffers[0].data();
```

Comment:
You should call `data<inner_value_type>` here, to be sure you cast the buffer into the right type.